assignment5/assignment5.cpp

```diff
@@ -12,33 +12,45 @@ class UnknownLanguage{
 	std::vector<std::string> dict;
 	void DictionaryToGraph(){
 		graph.clear();
-		for(int i=0; i<dict.size()-1; i++){
-			for(int j=0; j<dict[i].size(); j++){
-
-				if(j < dict[i+1].size() && 
-					dict[i].substr(0, j) == dict[i+1].substr(0, j)){
-
+		bool skip = false;
+		for(int i = 0; i+1<dict.size(); i++){
+			skip = false;
+			for(int j = 0; j<dict[i].size(); j++){
+				if(!skip && j < dict[i+1].size()){
 					if(dict[i][j]!=dict[i+1][j]){
 						graph[dict[i][j]].out_edges.push_back(dict[i+1][j]);
```

Comment:
Forgot to mention before, but using 'string' as a generic data structure is a common anti-pattern. Using it as a `vector<char>` is borderline, fell free to leave it as it is now, but I'll try to avoid it if possible.